removeDummies.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/python
+# The following script will take a 4D nifti images and will remove the requested number of volumes
+# in the 4th dimension
+
+# import system and OS
+import sys
+import os
+
+# Get the input from  the shell/bash script and assign them as variables
+print ('Running the script: '+sys.argv[0]) # prints python_script.py
+niftiFile = sys.argv[1]               # display the filename
+numDummies = int(sys.argv[2])         # display the number of dummies to remove
+deleteOriginal = int(sys.argv[3])     # Delete original nii.gz (with dummies) :1/0
+
+def removeDummies(niftiFile,numDummies,showHeader=0):
+
+    import nibabel as nib               # import nibabel
+    img = nib.load(niftiFile)           # load the nifti image
+    numDummies = numDummies             # Get number of dummies
+
+    path, file = os.path.split(niftiFile) # split the full path of the file
+    print(path)                           # print path name
+    print(file)                           # print file name
+
+    # read the header information from the nifti image
+    header = img.header
+    if showHeader:         # if header should be displayed
+        print('Header before removing dummies')
+        print(header)
+
+    # Get the affine matrix
+    affine = img.affine
+    image_data = img.get_fdata()    # Get the 4D matrix with the voxel values
+    dimOriginal = image_data.shape  # Get the shape of the matrix (Original 4D dimensions)
+
+    # Remove the requested dummies in the 4th dimensions
+    image_data = (image_data[:,:,:,numDummies:])
+    image_data = image_data.astype(int)      # Change the type of numbers to int
```

Comment:
for an unknown reason (to me) when the nifti images were read the numbers were who were originally integers and converted to float, and that lead to the output image to have different numbers (usually decimals) from the original  image. 
In this step, I am changing the type of the numbers to int before saving and that seems to solve the issue. 
When you try the script, please as a double check, see if the numbers are identical and the process didnt alter the voxel values.

Reply:
I checked that too and it is fine on my side